src/handlers/pricing/action.ts

```diff
@@ -37,10 +37,44 @@ export const pricingLabelLogic = async (): Promise<void> => {
   const minPriorityLabel = priorityLabels.length > 0 ? priorityLabels.reduce((a, b) => (calculateWeight(a) < calculateWeight(b) ? a : b)).name : undefined;
 
   const targetPriceLabel = getTargetPriceLabel(minTimeLabel, minPriorityLabel);
+
   if (targetPriceLabel) {
-    if (labels.map((i) => i.name).includes(targetPriceLabel)) {
-      logger.info(`Skipping... already exists`);
+    if (labels.map((i) => i.name).includes("Price")) {
+      if (!labels.map((i) => i.name).includes(targetPriceLabel)) {
+        // get all issue events of type "labeled" and the event label includes Price
+        const events = await getAllIssueEvents();
+        if (!events) return;
+        const labeledEvents: typeof events = [];
+        const labeledPriceEvents: typeof events = [];
+        events.forEach((event) => {
+          if (event.event === "labeled") {
+            labeledEvents.push(event);
+          }
+        });
```

Comment:
can we make this as a util function? like getLabeledEvents or sth. We need to make them as much as reusable.

Reply:
After I come back from school will do